flake.nix

```diff
@@ -0,0 +1,108 @@
+{
+  description = "slugger: Clean URI slugs for Haskell";
+
+  nixConfig.bash-prompt = "[nix]λ ";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+
+    flake-utils = {
+      url = "github:numtide/flake-utils";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+  };
+
+  outputs = { flake-utils, nixpkgs, self }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        config = {};
+
+        compilerVersion = "884";
+        compiler = "ghc" + compilerVersion;
+        overlays = [
+          (final: prev: {
+            haskell-language-server = prev.haskell-language-server.override {
+              supportedGhcVersions = [ compilerVersion ];
+            };
+
+            myHaskellPackages = prev.haskell.packages.${compiler}.override {
+              overrides = hpFinal: hpPrev: {
+                slugger = hpPrev.callCabal2nix "slugger" ./. {};
+
+                hspec = hpPrev.callHackage "hspec" "2.8.1" {}; # see slugger.cabal build-depends
+                hspec-core = hpPrev.callHackage "hspec-core" "2.8.1" {}; # because hspec
+                hspec-discover = hpPrev.callHackage "hspec-discover" "2.8.1" {}; # because hspec
+              };
+            };
+          })
+        ];
+
+        pkgs = import nixpkgs { inherit config overlays system; };
+      in rec {
+        packages = with pkgs.myHaskellPackages; [ slugger ];
+
+        devShell = pkgs.myHaskellPackages.shellFor {
+          packages = p: [
+            p.slugger
+          ];
+
+          buildInputs = with pkgs.myHaskellPackages; [
+            slugger
```

Comment:
In my experience, this sometimes is a hinderence; when you are on a directory with a broken source code, you can not enter `nix-shell` since it'll try to build the project. So I tend to not put the project itself to the `shell.nix`. In this case, it should be easy to do `cabal run` instead.

Reply:
I experienced that a few times. That's a good call

Reply:
I'm not sure what the problem is, but if I remove this, cabal can't find `text-icu`, and I can't run `cabal update` because of an `openFile` not found error... I'll get the other things going and circle back

Reply:
Maybe I need to call cabal 2 nix or something...........